server/src/main/java/io/deephaven/server/arrow/ArrowFlightUtil.java

```diff
@@ -326,50 +338,123 @@ public void onNext(final InputStream request) {
+        private void serviceDoGetRequest(BarrageProtoUtil.MessageInfo message) {
+            final DoGetRequest doGetRequest = DoGetRequest
+                    .getRootAsDoGetRequest(message.app_metadata.msgPayloadAsByteBuffer());
+
+            final SessionState.ExportObject<BaseTable> parent =
+                    ticketRouter.resolve(session, doGetRequest.ticketAsByteBuffer(), "ticket");
+
+            final BarrageSubscriptionRequest subscriptionRequest = BarrageSubscriptionRequest
+                    .getRootAsBarrageSubscriptionRequest(message.app_metadata.msgPayloadAsByteBuffer());
+
+            session.nonExport()
+                    .require(parent)
+                    .onError(listener)
+                    .submit(() -> {
+                        final BaseTable table = parent.get();
+
+                        // Send Schema wrapped in Message
+                        final FlatBufferBuilder builder = new FlatBufferBuilder();
+                        final int schemaOffset = BarrageUtil.makeSchemaPayload(builder, table.getDefinition(),
+                                table.getAttributes());
+                        builder.finish(MessageHelper.wrapInMessage(builder, schemaOffset,
+                                org.apache.arrow.flatbuf.MessageHeader.Schema));
+                        final ByteBuffer serializedMessage = builder.dataBuffer();
+
+                        // leverage the stream generator SchemaView constructor
+                        final BarrageStreamGenerator.SchemaView schemaView =
+                                new BarrageStreamGenerator.SchemaView(serializedMessage);
+
+                        // push the schema to the listener
+                        listener.onNext(schemaView);
+
+                        // collect the viewport and columnsets (if provided)
+                        final boolean hasColumns = subscriptionRequest.columnsVector() != null;
+                        final BitSet columns =
+                                hasColumns ? BitSet.valueOf(subscriptionRequest.columnsAsByteBuffer()) : null;
+
+                        final boolean hasViewport = subscriptionRequest.viewportVector() != null;
+                        final RowSet viewport =
+                                hasViewport ? BarrageProtoUtil.toIndex(subscriptionRequest.viewportAsByteBuffer())
+                                        : null;
+
+                        // get ourselves some data!
+                        final BarrageMessage msg = ConstructSnapshot.constructBackplaneSnapshotInPositionSpace(this,
+                                table, columns, viewport);
+                        msg.modColumnData = new BarrageMessage.ModColumnData[0]; // actually no mod column data for
+
+                        // translate the viewport to keyspace and make the call
+                        try (final BarrageStreamGenerator bsg = new BarrageStreamGenerator(msg);
+                                final RowSet keySpaceViewport =
```

Comment:
Unless I very much misunderstand, it is flat when sent over the wire - this index is just needed for the getSubView to know what contents to copy to the flight message

Reply:
```
        final int rowsAddedOffset;
        if (isSnapshot && !view.isInitialSnapshot) {
            // client's don't need/want to receive the full RowSet on every snapshot
            rowsAddedOffset = EmptyIndexGenerator.INSTANCE.addToFlatBuffer(metadata);
        } else {
            rowsAddedOffset = rowsAdded.addToFlatBuffer(metadata);
        }

        final int rowsRemovedOffset = rowsRemoved.addToFlatBuffer(metadata);
        final int shiftDataOffset = shifted.addToFlatBuffer(metadata);

        // Added Chunk Data:
        int addedRowsIncludedOffset = 0;
        if (view.isViewport()) {
            addedRowsIncludedOffset = rowsIncluded.addToFlatBuffer(view.keyspaceViewport, metadata);
        }
```
This code shows how the DoGet handling is replacing `rowsAdded` with an empty RowSet (not flat, but empty, even more concise). That is, in `FlightServiceGrpcImpl` it is somewhat falsely specifying that the SubView is not an initial snapshot.
`rowsIncluded` is not getting the same treatment, because the DoGet use case doesn't specify viewports.
We likely want to do something similar for our new request. Maybe time to add a boolean option "don't send row sets"?
Or maybe we can skip metadata altogether?